src/tools/w3dviewer/datatreeview.cpp

```diff
@@ -0,0 +1,680 @@
+                texture->Get_Name(),
+                m_imageListIDs[ICON_MATERIAL],
+                m_imageListIDs[ICON_MATERIAL],
+                0,
+                0,
+                0,
+                m_categoryTreeItems[CATEGORY_MATERIAL],
+                TVI_SORT);
+
+            AssetInfoClass *info = new AssetInfoClass(texture->Get_Name(), ASSET_TYPE_TEXTURE, nullptr, texture);
+            GetTreeCtrl().SetItem(newitem, TVIF_PARAM, nullptr, 0, 0, 0, 0, (LPARAM)info);
+        }
+    }
+}
+
+void CDataTreeView::AddAnimations()
+{
+    AssetIterator *iterator = W3DAssetManager::Get_Instance()->Create_HAnim_Iterator();
+
+    if (iterator != nullptr) {
+        for (iterator->First(); !iterator->Is_Done(); iterator->Next()) {
+            HAnimClass *anim = W3DAssetManager::Get_Instance()->Get_HAnim(iterator->Current_Item_Name());
+
+            for (HTREEITEM i = FindFirstItem(m_categoryTreeItems[CATEGORY_HIERARCHY], anim->Get_HName()); i != nullptr;
+                 i = FindNextItem(i, anim->Get_HName())) {
+                if (FindItemByName(i, iterator->Current_Item_Name()) == nullptr) {
+                    HTREEITEM newitem = GetTreeCtrl().InsertItem(TVIF_SELECTEDIMAGE | TVIF_TEXT | TVIF_IMAGE,
+                        iterator->Current_Item_Name(),
+                        m_imageListIDs[ICON_ANIMATION],
+                        m_imageListIDs[ICON_ANIMATION],
+                        0,
+                        0,
+                        0,
+                        i,
+                        TVI_SORT);
+
+                    AssetInfoClass *info =
+                        new AssetInfoClass(iterator->Current_Item_Name(), ASSET_TYPE_ANIMATION, nullptr, nullptr);
+                    GetTreeCtrl().SetItem(newitem, TVIF_PARAM, nullptr, 0, 0, 0, 0, (LPARAM)info);
+                }
+            }
+
+            for (HTREEITEM i = FindFirstItem(m_categoryTreeItems[CATEGORY_AGGREGATE], anim->Get_HName()); i != nullptr;
+                 i = FindNextItem(i, anim->Get_HName())) {
+                if (FindItemByName(i, iterator->Current_Item_Name()) == nullptr) {
+                    HTREEITEM newitem = GetTreeCtrl().InsertItem(TVIF_SELECTEDIMAGE | TVIF_TEXT | TVIF_IMAGE,
+                        iterator->Current_Item_Name(),
+                        m_imageListIDs[ICON_ANIMATION],
+                        m_imageListIDs[ICON_ANIMATION],
+                        0,
+                        0,
+                        0,
+                        i,
+                        TVI_SORT);
+
+                    AssetInfoClass *info =
+                        new AssetInfoClass(iterator->Current_Item_Name(), ASSET_TYPE_ANIMATION, nullptr, nullptr);
+                    GetTreeCtrl().SetItem(newitem, TVIF_PARAM, nullptr, 0, 0, 0, 0, (LPARAM)info);
+                }
+            }
+
+            for (HTREEITEM i = FindFirstItem(m_categoryTreeItems[CATEGORY_HLOD], anim->Get_HName()); i != nullptr;

```

Comment:
What is the difference in code of these 3 for loops other than the different `m_categoryTreeItems` ? Can this be refactored into 1 sub function and then called 3 times here, reducing source code size?